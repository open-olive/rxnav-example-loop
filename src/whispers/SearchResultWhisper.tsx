import { React } from '@oliveai/ldk';

import { DrugProperties } from '../aptitudes/network/RxNavClient';
import DrugResult from '../components/DrugResult';

// Expect the search term and results to be given as props for rendering
interface SearchResultWhisperProps {
  searchText: string;
  searchResults: DrugProperties[];
}
function SearchResultWhisper({ searchText, searchResults }: SearchResultWhisperProps) {
  const NUM_UNCOLLAPSED_RESULTS = 3;

  return (
    <oh-whisper
      label={`RxNav Drug Lookup - ${searchResults.length} results for "${searchText}"`}
      onClose={null}
    >
      {searchResults.slice(0, NUM_UNCOLLAPSED_RESULTS).map((drug, i) => (
        // Custom component used to render each result
        <DrugResult key={drug.rxcui} drug={drug} divider={i > 0} />
      ))}
      {searchResults.length > NUM_UNCOLLAPSED_RESULTS && (
        // If we have more than NUM_UNCOLLAPSED_RESULTS (3) results, wrap the rest in a closed collapse box
        <oh-collapse-box open={false}>
          {searchResults.slice(NUM_UNCOLLAPSED_RESULTS).map((drug) => (
            <DrugResult key={drug.rxcui} drug={drug} />
          ))}
        </oh-collapse-box>
      )}
    </oh-whisper>
  );
}

export default SearchResultWhisper;
