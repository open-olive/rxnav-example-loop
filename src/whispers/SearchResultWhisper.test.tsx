import { React } from '@oliveai/ldk';
import { create } from 'react-test-renderer';

import SearchResultWhisper from './SearchResultWhisper';

describe('SearchResultWhisper', () => {
  // Create a quick dummy array of search results for all tests
  const mockSearchResults = Array.from(Array(10).keys(), (_, i) => ({
    name: `foo ${i + 1}`,
    synonym: 'Synonym',
    rxcui: `RXCUI${i + 1}`,
    tty: 'TTY',
    language: 'Language',
    suppress: 'Suppress',
    umlscui: '',
  }));

  it('should render correctly for less than four results', () => {
    const component = create(
      <SearchResultWhisper searchText="foo" searchResults={mockSearchResults.slice(0, 3)} />
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly for four or more results', () => {
    const component = create(
      <SearchResultWhisper searchText="foo" searchResults={mockSearchResults} />
    );

    expect(component).toMatchSnapshot();
  });
});
