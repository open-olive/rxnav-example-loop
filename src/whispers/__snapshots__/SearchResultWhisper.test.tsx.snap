// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultWhisper should render correctly for four or more results 1`] = `
<oh-whisper
  label="RxNav Drug Lookup - 10 results for \\"foo\\""
  onClose={null}
>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-message
      header="foo 1"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI1"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-divider
      layout={
        Object {
          "paddingBottom": "small",
        }
      }
    />
    <oh-message
      header="foo 2"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI2"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-divider
      layout={
        Object {
          "paddingBottom": "small",
        }
      }
    />
    <oh-message
      header="foo 3"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI3"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
  <oh-collapse-box
    open={false}
  >
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 4"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI4"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 5"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI5"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 6"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI6"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 7"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI7"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 8"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI8"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 9"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI9"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
    <oh-box
      direction="vertical"
      justifyContent="left"
    >
      <oh-divider
        layout={
          Object {
            "paddingBottom": "small",
          }
        }
      />
      <oh-message
        header="foo 10"
      />
      <oh-list-pair
        copyable={true}
        label="Synonym"
        style="none"
        value="Synonym"
      />
      <oh-list-pair
        copyable={true}
        label="RXCUI"
        style="none"
        value="RXCUI10"
      />
      <oh-list-pair
        copyable={true}
        label="TTY"
        style="none"
        value="TTY"
      />
      <oh-list-pair
        copyable={true}
        label="Language"
        style="none"
        value="Language"
      />
      <oh-list-pair
        copyable={true}
        label="Suppress"
        style="none"
        value="Suppress"
      />
    </oh-box>
  </oh-collapse-box>
</oh-whisper>
`;

exports[`SearchResultWhisper should render correctly for less than four results 1`] = `
<oh-whisper
  label="RxNav Drug Lookup - 3 results for \\"foo\\""
  onClose={null}
>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-message
      header="foo 1"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI1"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-divider
      layout={
        Object {
          "paddingBottom": "small",
        }
      }
    />
    <oh-message
      header="foo 2"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI2"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
  <oh-box
    direction="vertical"
    justifyContent="left"
  >
    <oh-divider
      layout={
        Object {
          "paddingBottom": "small",
        }
      }
    />
    <oh-message
      header="foo 3"
    />
    <oh-list-pair
      copyable={true}
      label="Synonym"
      style="none"
      value="Synonym"
    />
    <oh-list-pair
      copyable={true}
      label="RXCUI"
      style="none"
      value="RXCUI3"
    />
    <oh-list-pair
      copyable={true}
      label="TTY"
      style="none"
      value="TTY"
    />
    <oh-list-pair
      copyable={true}
      label="Language"
      style="none"
      value="Language"
    />
    <oh-list-pair
      copyable={true}
      label="Suppress"
      style="none"
      value="Suppress"
    />
  </oh-box>
</oh-whisper>
`;
